src/platforms/stm32/serialno.c

```diff
@@ -23,44 +23,41 @@
 char *serial_no_read(char *s)
 {
 #if DFU_SERIAL_LENGTH == 9
-	int i;
-	volatile uint32_t *unique_id_p = (volatile uint32_t *)DESIG_UNIQUE_ID_BASE;
-	uint32_t unique_id = *unique_id_p +
-			*(unique_id_p + 1) +
-			*(unique_id_p + 2);
+	const volatile uint32_t *const unique_id_p = (volatile uint32_t *)DESIG_UNIQUE_ID_BASE;
```

Comment:
also the volatile cast does nothing here i think

Reply:
The volatile guarantees we generate the instructions to read the value, which is correct.. it shouldn't be hurting perf as we only read each location once. As above, the second const makes it so we're reading a constant value from a constant pointer - this prevents us accidentally writing back to the location or changing the value and promotes certain optimisations that are otherwise prohibited.

Reply:
it would hurt perf if a location is read multiple times - that is why we point it out as not being a problem in this instance. We're about to push again with cleanup for this inconsistency :)

Reply:
yes, volatile would tell the compiler to actually read the variable every access, where it might've optimised otherwise, but thats why I was thinking i wouldnt be necessary here